deepchem/models/wandblogger.py

```diff
@@ -0,0 +1,115 @@
+                ' install it with `pip install wandb`.'
+            )
+
+        if mode == "offline" and log_model:
+            raise Exception(
+                f'Providing log_model={log_model} and mode={mode} is an invalid configuration'
+                ' since model checkpoints cannot be uploaded in offline mode.\n'
+                'Hint: Set `mode="online"` to log your model.'
+            )
+        #Check for datasets and metrics
+        if (datasets is None) or (not datasets):
+            raise Exception(
+                'No datasets are provided: `datasets` is not set or is empty.'
+            )
+        if (metrics is None) or (not metrics):
+            raise Exception(
+                'No metrics are provided: `metrics` is not set or is empty'
+            )
+
+
+        self.base_model = None # will be set in KerasModel init
+        self.datasets = datasets
+        #if datasets is in list format, convert to dictionary
+        if isinstance(datasets, list):
+            self.datasets = {"dataset_" + str(k): v for k, v in enumerate(self.datasets)}
+
+        #identify the train dataset for epoch calculation purposes
+        # assign first dataset as train set by default
+        if isinstance(train_dataset_id, int) and isinstance(datasets, list):
+            self.train_dataset_name = "dataset_" + str(train_dataset_id)
+        elif isinstance(train_dataset_id, str) and isinstance(datasets, dict):
+            self.train_dataset_name = train_dataset_id
+        else:
+            self.train_dataset_name = list(self.datasets.keys())[0]
+            logger.warning("WandbLogger: 'train_dataset_id' not provided or invalid. "
+                           "It must be a str/int for the dataset at key/index of 'datasets'."
+                           "Train dataset will default to first dataset in 'datasets'.")
+
+        self.metrics = metrics
+
+        self.log_model = log_model
+        self.save_dir = save_dir
+
+        # set wandb init arguments
+        self.wandb_init_params = dict(
+            name=name,
+            project=project,
+            entity=entity,
+            mode=mode,
+            id=id,
+            dir=save_dir,
+            resume='allow',
+            anonymous=anonymous
+        )
+        self.wandb_init_params.update(**kwargs)
+
+        self.wandb = wandb.init(**self.wandb_init_params) if wandb.run is None else wandb.run
+
+    def save_model(self, model):
+        '''
+        Not started yet
+        '''
+        #model is a tf.keras.Model
+        return None
+
+    def log_data(self, data, step):
+        self.wandb.log(data)
+
+    def calculate_metrics(self, model):
+        #model is a Deepchem KerasModel
+        all_metrics = dict({})
+        for name, dataset in self.datasets.items():
+            #Calculate and log metrics on each dataset
+            scores = model.evaluate(dataset, self.metrics)
+            scores = {'metrics/' + k + '_' + name: v for k, v in scores.items()}
```

Comment:
Assuming this is only called during validation lets name this the metrics as "valid/...". If its not we should pass a param to `calculate_metrics` to indicate if we are training or evaluating

Reply:
The outer for loop goes through each dataset and calculates metrics for each.

Reply:
Will need to rename after datasets + single logging function is done.